cmake/WarningsAsErrors.cmake

```diff
@@ -0,0 +1,31 @@
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_SIMULATE_ID MATCHES
+                                             "MSVC"
+)
+    target_compile_options(${PROJECT_NAME} PRIVATE /W4 /WX)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    target_compile_options(
+        ${PROJECT_NAME}
+        PRIVATE -Wall
```

Comment:
FWIW, I think we should _either_ have -Wall _or_ -Werror ... having both on puts the compilation at the mercy of the compiler version, as GCC 27 might add a brand new warning and suddenly everyone's build starts breaking.

If we have -Wall, then we should not have -Werror except for those warnings that we have explicitly fixed and promised do not exist.